README.md

```diff
@@ -103,6 +103,52 @@ make manifests
 
 More information can be found via the [Kubebuilder Documentation](https://book.kubebuilder.io/introduction.html)
 
+## Logging
+Logs are following the general guidelines: 
+- `logger.Info()` describe a high-level state of the resource such as creation, deletion and which reconciliation path was taken.  
+- `logger.Error()` describe only those errors that are not returned in the result of the reconciliation. If error is occurred there should be only one error message. 
+- `logger.V(1).Info()` debug level logs to give information about every change or event caused by the resource as well as every update of the resource.
+
+The `--zap-devel` argument will enable debug level logs for the output. Otherwise, all `V()` logs are ignored.
+
+### Common metadata 
+Not exhaustive list of metadata for DNSRecord controller:
+- `level` - logging level. Values are: `info`,`debug` or `error`
+- `ts` - timestamp
+- `logger` - logger name
+- `msg`
+- `controller` and `controllerKind` - controller name, and it's kind respectively to output the log
+- `DNSRecord` - name and namespace of the DNS Record CR that is being reconciled
+- `reconcileID`
+- `ownerID` - ID the of owner of the DNS Record 
+- `txtPrefix`/`txtSuffix` - prefix and suffix of the TXT record in provider. 
+- `zoneEndpoints` - endpoints that exist in the provider
+- `specEdnoinds` - endpoints defined in the spec
+- `statusEndpoints` - endpoints that were processed previously
```

Comment:
nit: I would have kept this list down to the important pieces of data that are attached to all requests and not include some that are logged in one log statement. zoneEndpoint, specEndpoints and statusEndpoints txtPrefix are not goign to be on all requests.